BSPConvert.Lib/Source/EntityConverter.cs

```diff
@@ -597,23 +612,192 @@ private void ConvertTriggerTargetsRecursive(Entity trigger, Entity entity, float
 						ConvertTargetPrintTrigger(trigger, target, delay);
 						break;
 					case "target_speed":
-						FireTargetSpeedOnOutput(trigger, target, "OnStartTouch", delay);
+						FireTargetSpeedOnOutput(trigger, target, "OnTrigger", delay);
 						break;
 					case "target_push":
 						ConvertTargetPushTrigger(trigger, target, delay);
 						break;
 					case "target_remove_powerups":
-						SetHasteOnOutput(trigger, "0", "OnStartTouch", delay);
-						SetQuadOnOutput(trigger, "0", "OnStartTouch", delay);
+						SetHasteOnOutput(trigger, "0", "OnTrigger", delay);
+						SetQuadOnOutput(trigger, "0", "OnTrigger", delay);
 						break;
 					case "func_door":
-						OpenDoorOnOutput(trigger, target, "OnStartTouch", delay);
+						OpenDoorOnOutput(trigger, target, "OnTrigger", delay);
+						break;
+					case "target_fragsFilter":
+						ConvertFragsFilter(trigger, target, "OnTrigger", delay);
+						break;
+					case "target_score":
+						ConvertTargetScore(trigger, target, "OnTrigger", delay);
 						break;
 				}
 
 				visited.Add(target);
-				ConvertTriggerTargetsRecursive(trigger, target, delay, visited);
+				if (target.ClassName != "logic_relay") // logic_relay moves the next target's inputs to a separate entity instead, break from the loop
+					ConvertTriggerTargetsRecursive(trigger, target, delay, visited);
+			}
+		}
+
+		private void ConvertTargetTeleporter(Entity entity, Entity targetTeleporter, string output, float delay)
+		{
+			var targets = GetTargetEntities(targetTeleporter);
+			foreach (var target in targets)
+			{
+				target.ClassName = "point_teleport";
+				target["target"] = "!player";
+
+				// TODO: Implement teleport velocity modes, fix teleport origin
+
+				var connection = new Entity.EntityConnection()
+				{
+					name = output,
+					target = target.Name,
+					action = "Teleport",
+					param = null,
+					delay = delay,
+					fireOnce = -1
+				};
+				entity.connections.Add(connection);
+			}
+		}
+
+		private void ConvertTargetScore(Entity entity, Entity targetScore, string output, float delay)
+		{
+			if (sourceEntities.FirstOrDefault(x => x.ClassName == "math_counter") == null) // Check if math_counter exists
+				CreateMathCounter();
+
+			if (!float.TryParse(targetScore["count"], out var count))
+				count = 1;
+
+			ModifyMathCounter(entity, output, "Add", $"{count}", delay);
+		}
+
+		private Entity CreateLogicCase()
+		{
+			var logicCase = new Entity();
+			logicCase.ClassName = "logic_case";
+			logicCase.Name = "logiccase";
+
+			for (var i = 1; i <= 16; i++) // Logic_case supports 16 different outputs
+			{
+				var caseNum = $"case{i:D2}";
+				logicCase[$"{caseNum}"] = $"{i}";
```

Comment:
I think the string interpolation is unnecessary here.
`logicCase[caseNum] = i;`. If that doesn't work, then just add `.ToString()`